### PR TITLE
Implemented time out report to log cluster status and condition infor…

### PR DESCRIPTION
**Pull request by lscalabrini01**
Implemented time out report to log cluster status and condition information to the provisioning tests

- Implemented TimeoutRKEReport function
- Implemented TimeoutClusterReport function
- Refactor watch function to use TimeoutError and WatchConnectionError variables
- Change VerifyRKE1Cluster to use TimeoutRKEReport function
- Change VerifyCluster to use TimeoutClusterReport function
- Change VerifyHostedCluster to use TimeoutRKEReport function

### CODE REVIEW

File: extensions/reports/timeout_report.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package reports
+
+import (
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/sirupsen/logrus"
+)
+
+func TimeoutReport(cluster *management.Cluster, err error) {
```
Comment (igomez06) [inline extensions/reports/timeout_report.go:9]: Can you add go doc comment here?

**Review by igomez06 — COMMENTED** [review]
looks great! Just one comment. And we can we get backports to release/v2.8 and release/v2.7?

File: extensions/reports/timeout_report.go (hunk 1 above)
Comment (caliskanugur) [reply extensions/reports/timeout_report.go:9]: Agreed,  also I have a question: will this work for v1 clusters?

**Review by caliskanugur — COMMENTED** [review]
Looks awesome! just one question

File: extensions/reports/timeout_report.go (hunk 1 above)
Comment (igomez06) [reply extensions/reports/timeout_report.go:9]: Good point Ugur. You might need to have two functions. One for rke1 and one rke2/k3s

Comment (lscalabrini01) [reply extensions/reports/timeout_report.go:9]: Yes, I can

Comment (lscalabrini01) [reply extensions/reports/timeout_report.go:9]: I'll crate these new functions

**Review by floatingman — DISMISSED** [review]
Looks good to me

**Review by floatingman — APPROVED** [review]
LGTM

**Pull request merged**